src/test/java/uk/gov/companieshouse/authcodenotification/service/AuthCodeNotificationServiceTest.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.companieshouse.authcodenotification.service;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.api.model.update.OverseasEntityDataApi;
+import uk.gov.companieshouse.authcodenotification.exception.ServiceException;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class AuthCodeNotificationServiceTest {
+
+    private static final String REQUEST_ID = "abc";
+
+    private static final String COMPANY_NUMBER = "OE123456";
+
+    private static final String TEST_EMAIL = "test@oe.com";
+
+    @InjectMocks
+    private AuthCodeNotificationService authCodeNotificationService;
+
+    @Mock
+    private PrivateDataRetrievalService privateDataRetrievalService;
+
+    @Test
+    void testSuccessfulRetrieveAndSend() throws ServiceException {
+        OverseasEntityDataApi overseasEntityDataApi = new OverseasEntityDataApi();
+        overseasEntityDataApi.setEmail(TEST_EMAIL);
+        when(privateDataRetrievalService.getOverseasEntityData(REQUEST_ID, COMPANY_NUMBER)).thenReturn(overseasEntityDataApi);
+        authCodeNotificationService.sendAuthCodeEmail(REQUEST_ID, COMPANY_NUMBER);
+        verify(privateDataRetrievalService, times(1)).getOverseasEntityData(REQUEST_ID, COMPANY_NUMBER);
+    }
+
+    @Test
+    void testBlankEmailExceptionThrown() throws ServiceException {
```

Comment:
'testExceptionThrownWhenEmailIsBlankOrNotFound'

Reply:
Method name changed.

Reply:
Thanks for the change but think you might have done that based on an older version of the comment. Name needs to be specific to the test you're checking - in this case it's a null email being set but we should also have a test for a blank email being set (or a a string containing just whitespace).

Reply:
Separated the tests